tests/unit/test_client.py

```diff
@@ -97,15 +103,17 @@ def _constructor_test_helper(
                 client._client_options.api_endpoint,
                 expected_client_options.api_endpoint,
             )
+        if expected_query_options is not None:
+            self.assertEqual(client._query_options, expected_query_options)
 
-    @mock.patch("google.cloud.spanner_v1.client.os.getenv")
+    @mock.patch("google.cloud.spanner_v1.client._get_spanner_emulator_host")
```

Comment:
Not sure why there's a diff for `_get_spanner_emulator_host` in this PR.

Reply:
Both emulator support and query options support rely on calling os.getenv which means it's more difficult to set up the mock. The `os.getenv` call for emulator is made in `_get_spanner_emulator_host` which means mocking that function instead is much more clear.